docs/content/latest/explore/indexes-constraints/primary-key-ysql.md

```diff
@@ -0,0 +1,159 @@
+</ul>
+
+The Primary Key constraint is a means to identify a specific row in a table uniquely via one or more columns. To define a primary key, you create a constraint that is, functionally, a [unique index](../indexes-1/#using-a-unique-index) applied to the table columns.
+
+## Syntax and examples
+
+- To run the examples below, follow these steps to create a local [cluster](/latest/quick-start/) or in [Yugabyte Cloud](/latest/yugabyte-cloud/cloud-connect/).
+
+- Use the [YSQL shell](/latest/admin/ysqlsh/) for local clusters, or [Connect using Cloud shell](/latest/yugabyte-cloud/cloud-connect/connect-cloud-shell/) for Yugabyte Cloud, to create a database and a table.
+
+### Primary key for a single column
+
+Most commonly, the primary key is added to the table when the table is created, as demonstrated by the following syntax:
+
+```sql
+CREATE TABLE (
+  column1 data_type PRIMARY KEY,
+  column2 data_type,
+  column3 data_type,
+  …
+);
+```
+
+The following example creates the `employee` table with `employee_no` as the primary key, which uniquely identifies an employee.
+
+```sql
+CREATE TABLE employees (
+  employee_no integer PRIMARY KEY,
+  name text,
+  department text
+);
+```
+
+### Primary key over multiple columns
+
+The following syntax can be used to define a primary key for more than one column:
+
+```sql
+CREATE TABLE (
+  column1 data_type,
+  column2 data_type,
+  column3 data_type,
+  …
+  PRIMARY KEY (column1, column2)
+);
+
+```
+
+The following example creates the `employees` table in which the primary key is a combination of `employee_no` and `name` columns:
+
+```sql
+CREATE TABLE employees (
+  employee_no integer,
+  name text,
+  department text,
+  PRIMARY KEY (employee_no, name)
+);
+```
+
+### CONSTRAINT
+
+YSQL assigns a default name in the format `tablename_pkey` to the primary key constraint. For example, the default name is `employees_pkey` for the `employees` table. If you need a different name, you can specify it using the `CONSTRAINT` clause, as per the following syntax:
+
+```sql
+CONSTRAINT constraint_name PRIMARY KEY(column1, column2, ...);
+```
+
+The following example demonstrates the use of `CONSTRAINT` to change the default name.
+
+```sql
+CONSTRAINT employee_no_pkey PRIMARY KEY(employee_no);
+```
+
+### ALTER TABLE
+
+- Use the `ALTER TABLE` statement to create a primary key on an existing table with following syntax:
+
+```sql
+ALTER TABLE table_name ADD PRIMARY KEY (column1, column2);
+```
+
+The following example creates the `employee` table first and then alters it to add a primary key on the `employee_no` column:
+
+```sql
+CREATE TABLE employees (
+  employee_no integer,
+  name text,
+  department text
+);
+
+ALTER TABLE employees ADD PRIMARY KEY (employee_no);
+```
+
+- The `ALTER TABLE` statement also allows you to add an auto-incremented primary key to an existing table by using the [SERIAL type](../../ysql-language-features/data-types/#serial-pseudotype), as per the following syntax:
```

Comment:
Shouldn't we recommend `bigint generated always as identity` rather `serial`? It has more possibilities, like setting the cache size.